### DISCUSSION TITLE
How to use websocket with socket.io?

### DISCUSSION THREAD

**Discussion by martinszeltins** [vueuse/vueuse#2085] [General]
There are some general examples with `useWebsocket` but how would you use it with `socket.io` for example? Can you use `useWebsocket` to connect / join a room etc.?

**Comment by curtgrimes** [comment]
[Socket.io is not a WebSocket implementation](https://socket.io/docs/v4/#what-socketio-is-not), so VueUse's `useWebSocket()` composable is not compatible with Socket.IO connections. But here are some resources that might be helpful in getting a Vue app talking with Socket.IO connections:

- https://github.com/MetinSeylan/Vue-Socket.io - A Vue + Socket.IO implementation. Appears to only support the options API style.
- https://stackoverflow.com/questions/71466297/how-to-use-vue-3-socket-io-with-composition-api-in-vue-js - A discussion about using Socket.IO directly in a Vue 3 app.
